### PR TITLE
fixes #42 - Adds pre-commit hook for flake8 and black

### LINKED ISSUE #42: Define the python linter
There are many different linters for python.
We should try to adopt one style and be using the same. Let's document that in the project.

flake8 is a wrapper around linters. so we can add whichever we think is interesting.

For example pylint would detect the non documented `class` (missing docstrings).

### PR DESCRIPTION
**Pull request by karlcow**
This will add pre-commit hook. I still need to figure out how to properly set the flake8 line length through the pre-commit.

### CODE REVIEW

**Comment by karlcow** [comment]
It seems to be working locally

```
% git commit -m 'issue #42 - adds rules for flake8' .
black................................................(no files to check)Skipped
flake8...............................................(no files to check)Skipped
[42/1 b79217b] issue #42 - adds rules for flake8
 1 file changed, 5 insertions(+)
 create mode 100644 .flake8
```

but now fails on CircleCi.
https://app.circleci.com/pipelines/github/webcompat/webcompat-metrics-server/11/workflows/986a19b2-13de-4701-819f-aa0af1a33fbd/jobs/197

```
#!/bin/bash -eo pipefail
. venv/bin/activate
flake8 --exclude=venv
nose2

./ochazuke/helpers.py:123:18: E126 continuation line over-indented for hanging indent

./ochazuke/helpers.py:124:58: E226 missing whitespace around arithmetic operator

./ochazuke/helpers.py:125:17: E126 continuation line over-indented for hanging indent

./ochazuke/helpers.py:140:18: E126 continuation line over-indented for hanging indent

./ochazuke/helpers.py:141:56: E226 missing whitespace around arithmetic operator

./ochazuke/helpers.py:142:17: E126 continuation line over-indented for hanging indent

./bin/weekly_total.py:64:17: E123 closing bracket does not match indentation of opening bracket's line


Exited with code exit status 1

CircleCI received exit code 1
```

**Comment by karlcow** [comment]
```
(env) ~/code/webcompat-metrics-server % git commit -m 'issue #42 - fixes flake8 issues' .
black....................................................................Failed
- hook id: black
- files were modified by this hook

reformatted /Users/karl/code/webcompat-metrics-server/config.py
reformatted /Users/karl/code/webcompat-metrics-server/bin/daily_total.py
reformatted /Users/karl/code/webcompat-metrics-server/bin/weekly_total.py
reformatted /Users/karl/code/webcompat-metrics-server/ochazuke/helpers.py
All done! ✨ 🍰 ✨
4 files reformatted.

flake8...................................................................Passed
(env) ~/code/webcompat-metrics-server % flake8 --exclude env                             
(env) ~/code/webcompat-metrics-server % git push origin 42/1                                    
Everything up-to-date
(env) ~/code/webcompat-metrics-server % git status
Sur la branche 42/1
Modifications qui ne seront pas validées :
  (utilisez "git add <fichier>..." pour mettre à jour ce qui sera validé)
  (utilisez "git restore <fichier>..." pour annuler les modifications dans le répertoire de travail)
	modifié :         bin/daily_total.py
	modifié :         bin/weekly_total.py
	modifié :         config.py
	modifié :         ochazuke/helpers.py

aucune modification n'a été ajoutée à la validation (utilisez "git add" ou "git commit -a")
(env) ~/code/webcompat-metrics-server % git commit -m 'issue #42 - fixes flake8 issues' .
black....................................................................Passed
flake8...................................................................Passed
[42/1 da6d0be] issue #42 - fixes flake8 issues
 4 files changed, 71 insertions(+), 55 deletions(-)
(env) ~/code/webcompat-metrics-server % git push origin 42/1                             
Énumération des objets: 15, fait.
Décompte des objets: 100% (15/15), fait.
Compression par delta en utilisant jusqu'à 8 fils d'exécution
Compression des objets: 100% (8/8), fait.
Écriture des objets: 100% (8/8), 1.46 Kio | 1.46 Mio/s, fait.
Total 8 (delta 6), réutilisés 0 (delta 0), réutilisés du pack 0
remote: Resolving deltas: 100% (6/6), completed with 6 local objects.
To github.com:webcompat/webcompat-metrics-server.git
   b79217b..da6d0be  42/1 -> 42/1
```

Seems to be working. 
What about CircleCI? YEAH CircleCI too.

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[flake8]
+ignore = E203, E266, E501, W503, F403, F401
+max-line-length = 79
```
[the remark this answers is not in the document]
Comment (karlcow) [reply .flake8:3]: hmm so what do we do, longer lines in `flake8` or adding a `pyproject.toml`. Life choices. 🤣

File: .flake8 [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+max-complexity = 18
```
[the remark this answers is not in the document]
Comment (karlcow) [reply .flake8:4]: no better insights just copy-pasta and it worked… so 

https://flake8.pycqa.org/en/3.8.3/user/options.html#cmdoption-flake8-max-complexity
and the current code is happy with 18 (or it seems). 
I honestly do not have strong opinion. 


Let's keep what you recommended.
* [E203](https://www.flake8rules.com/rules/E203.html) white space before `:`. black adds the space time to time.
* [W503](https://www.flake8rules.com/rules/W503.html) binary operator line breaking.


with this we get

```
(env) ~/code/webcompat-metrics-server % flake8 --max-complexity 18 --exclude env
./bin/weekly_total.py:70:80: E501 line too long (87 > 79 characters)
./bin/daily_total.py:65:80: E501 line too long (83 > 79 characters)
./bin/daily_total.py:83:80: E501 line too long (85 > 79 characters)
./ochazuke/helpers.py:127:80: E501 line too long (86 > 79 characters)
./ochazuke/helpers.py:146:80: E501 line too long (85 > 79 characters)
```

> very sophisticated team

Aren't we? 😛

File: .flake8 (hunk 1 above)
[the remark this answers is not in the document]
Comment (karlcow) [reply .flake8:3]: a couple of minutes later. 

Let's try 88 lines. Specifically it doesn't matter anymore that much in the sense or black is here to fix it. And people can focus on writing logic more than styling.

File: ochazuke/helpers.py [hunk 3]
```diff
@@ -113,31 +120,34 @@ def get_timeline_data(category, start, end):
...
     logging.info(msg)
-    reports_list = WeeklyTotal.query.filter(date_range).order_by(
-        WeeklyTotal.monday.asc()).all()
-    logging.info('REPORTS {}'.format(reports_list))
-    timeline = [{
-                 'count': report.count,
-                 'timestamp': report.monday.isoformat()+'Z'
-                } for report in reports_list]
+    reports_list = (
+        WeeklyTotal.query.filter(date_range).order_by(WeeklyTotal.monday.asc()).all()
```
[the remark this answers is not in the document]
Comment (karlcow) [reply ochazuke/helpers.py:146]: addressed by changing line length

File: bin/weekly_total.py [hunk 4]
```diff
@@ -59,15 +60,15 @@ def main():
...
+            msg_tmp = "Successfully wrote count for {} in WeeklyTotal table."
+            msg = (msg_tmp).format(monday)
             LOGGER.info(msg)
         # Catch error and attempt to recover by resetting staged changes.
         except sqlalchemy.exc.SQLAlchemyError as error:
             db.session.rollback()
-            msg = ("Yikes! Failed to write data for {week} in "
-                   "WeeklyTotal table: {err}").format(week=monday, err=error)
+            msg = (
+                "Yikes! Failed to write data for {week} in " "WeeklyTotal table: {err}"
```
[the remark this answers is not in the document]
Comment (karlcow) [reply bin/weekly_total.py:70]: addressed by changing line length

File: bin/daily_total.py [hunk 5]
```diff
@@ -62,26 +62,26 @@ def main():
         issue_count = get_issue_count(json_response)
         if not issue_count:
             # On a second failure, log an error
-            msg = "Daily count failed for {yesterday}!".format(
-                yesterday=yesterday
-            )
+            msg = "Daily count failed for {yesterday}!".format(yesterday=yesterday)
```
[the remark this answers is not in the document]
Comment (karlcow) [reply bin/daily_total.py:65]: addressed by changing line length

File: bin/weekly_total.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (laghee) [reply bin/weekly_total.py:70]: Oh, wait, should we fix this silly line before merging, though?

**Comment by karlcow** [comment]
Yikes "yikes".

**Comment by laghee** [comment]
Eeek! Such timely responsiveness! 😆

**Pull request merged**